src/lazyload-image.directive.ts

```diff
@@ -1,44 +1,61 @@
 })
-export class LazyLoadImageDirective {
-    @Input('lazyLoad') lazyImage;   // The image to be lazy loaded
-    @Input() errorImage: string;    // The image to be displayed if lazyImage load fails
-    @Input() scrollTarget = window; // Chnage the node we should listen for scroll events on, default is window
-    @Input() scrollObservable;      // Pass your own scroll emitter
-    @Input() offset: number;        // The number of px a image should be loaded before it is in view port
-    elementRef: ElementRef;
-    ngZone: NgZone;
-    scrollSubscription;
-
-    constructor(el: ElementRef, ngZone: NgZone) {
-        this.elementRef = el;
-        this.ngZone = ngZone;
-    }
+export class LazyLoadImageDirective implements OnChanges, OnDestroy {
+  @Input('lazyLoad') lazyImage;   // The image to be lazy loaded
+  @Input() errorImage: string;    // The image to be displayed if lazyImage load fails
+  @Input() scrollTarget; // Change the node we should listen for scroll events on, default is window
+  @Input() scrollObservable;      // Pass your own scroll emitter
+  @Input() offset: number;        // The number of px a image should be loaded before it is in view port
+  elementRef: ElementRef;
+  ngZone: NgZone;
+  scrollSubscription;
+  window;
+
+  constructor(el: ElementRef, ngZone: NgZone, window: WindowService) {
+    this.elementRef = el;
+    this.ngZone = ngZone;
+    // Get window object, fallback for Universal
+    this.window = window.nativeWindow || (<any>global);
+    this.scrollTarget = this.window;
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
 
-    ngAfterContentInit() {
+    let chngImg = changes['lazyImage'];
+    if (chngImg) {
+      let currImg = chngImg.currentValue;
+      let prevImg = chngImg.previousValue;
+
+      if (currImg !== prevImg) {
         this.ngZone.runOutsideAngular(() => {
-            if (this.scrollObservable) {
-                this.scrollSubscription = this.scrollObservable
-                    .startWith('')
-                    .let(lazyLoadImage(this.elementRef.nativeElement, this.lazyImage, this.errorImage, this.offset))
-                    .subscribe(() => {});
-            } else {
-                this.scrollSubscription = getScrollListener(this.scrollTarget)
-                    .let(lazyLoadImage(this.elementRef.nativeElement, this.lazyImage, this.errorImage, this.offset))
-                    .subscribe(() => {});
-            }
+          if (this.scrollObservable) {
+            this.scrollSubscription = this.scrollObservable
+              .startWith('')
+              .let(lazyLoadImage(this.elementRef.nativeElement, currImg, this.errorImage, this.offset, this.window))
+              .subscribe();
+          } else {
+            this.scrollSubscription = getScrollListener(this.scrollTarget)
+              .let(lazyLoadImage(this.elementRef.nativeElement, currImg, this.errorImage, this.offset, this.window))
+              .subscribe();
+          }
         });
+      }
```

Comment:
We need to unsubscribe to the old observable if we start a new one. We will otherwise have memory leeks and if we pass the same image element to the `lazyLoadImage`-function the previous image may be activated after the new images has been loaded. Meaning, we can't guarantee the loading order if we do not unsubscribe.